### ISSUE TITLE
[Bug] Invalid column types provided for the DbSchemaXmlGenerator

### ISSUE THREAD

**Issue by Bashev** [magento/magento2-phpstorm-plugin#626]
**Describe the bug** (*)

```
java.util.InputMismatchException: Invalid column types provided. Should be compatible with class com.magento.idea.magento2plugin.magento.packages.database.TableColumnTypes
	at com.magento.idea.magento2plugin.actions.generation.generator.DbSchemaXmlGenerator.generate(DbSchemaXmlGenerator.java:106)
	at com.magento.idea.magento2plugin.actions.generation.generator.FileGenerator.generate(FileGenerator.java:41)
	at com.magento.idea.magento2plugin.actions.generation.dialog.NewDbSchemaDialog.generateDbSchemaXmlFile(NewDbSchemaDialog.java:181)
	at com.magento.idea.magento2plugin.actions.generation.dialog.NewDbSchemaDialog.onOK(NewDbSchemaDialog.java:149)
	at com.magento.idea.magento2plugin.actions.generation.dialog.NewDbSchemaDialog.lambda$new$0(NewDbSchemaDialog.java:106)
	at java.desktop/javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1967)
	at java.desktop/javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2308)
	at java.desktop/javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:405)
	at java.desktop/javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:262)
	at java.desktop/javax.swing.AbstractButton.doClick(AbstractButton.java:369)
	at java.desktop/javax.swing.plaf.basic.BasicRootPaneUI$Actions.actionPerformed(BasicRootPaneUI.java:254)
	at java.desktop/javax.swing.SwingUtilities.notifyAction(SwingUtilities.java:1810)
	at java.desktop/javax.swing.JComponent.processKeyBinding(JComponent.java:2903)
	at java.desktop/javax.swing.KeyboardManager.fireBinding(KeyboardManager.java:311)
	at java.desktop/javax.swing.KeyboardManager.fireKeyboardAction(KeyboardManager.java:253)
	at java.desktop/javax.swing.JComponent.processKeyBindingsForAllComponents(JComponent.java:2996)
	at java.desktop/javax.swing.JComponent.processKeyBindings(JComponent.java:2988)
	at java.desktop/javax.swing.JComponent.processKeyEvent(JComponent.java:2865)
	at java.desktop/java.awt.Component.processEvent(Component.java:6429)
	at java.desktop/java.awt.Container.processEvent(Container.java:2263)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:5027)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4859)
	at java.desktop/java.awt.KeyboardFocusManager.redispatchEvent(KeyboardFocusManager.java:1962)
	at java.desktop/java.awt.DefaultKeyboardFocusManager.dispatchKeyEvent(DefaultKeyboardFocusManager.java:879)
	at java.desktop/java.awt.DefaultKeyboardFocusManager.preDispatchKeyEvent(DefaultKeyboardFocusManager.java:1158)
	at java.desktop/java.awt.DefaultKeyboardFocusManager.typeAheadAssertions(DefaultKeyboardFocusManager.java:1018)
	at java.desktop/java.awt.DefaultKeyboardFocusManager.dispatchEvent(DefaultKeyboardFocusManager.java:844)
	at com.intellij.ide.IdeKeyboardFocusManager.dispatchEvent(IdeKeyboardFocusManager.java:42)
	at java.desktop/java.awt.Component.dispatchEventImpl(Component.java:4908)
	at java.desktop/java.awt.Container.dispatchEventImpl(Container.java:2321)
	at java.desktop/java.awt.Window.dispatchEventImpl(Window.java:2784)
	at java.desktop/java.awt.Component.dispatchEvent(Component.java:4859)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:778)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:95)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:751)
	at java.desktop/java.awt.EventQueue$5.run(EventQueue.java:749)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:748)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:885)
	at com.intellij.ide.IdeEventQueue.dispatchKeyEvent(IdeEventQueue.java:801)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:748)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:441)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:825)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:440)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:794)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:492)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:117)
	at java.desktop/java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:190)
	at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:235)
	at java.desktop/java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:233)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.desktop/java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:233)
	at java.desktop/java.awt.Dialog.show(Dialog.java:1063)
	at java.desktop/java.awt.Component.show(Component.java:1732)
	at java.desktop/java.awt.Component.setVisible(Component.java:1679)
	at java.desktop/java.awt.Window.setVisible(Window.java:1032)
	at java.desktop/java.awt.Dialog.setVisible(Dialog.java:998)
	at com.magento.idea.magento2plugin.actions.generation.dialog.NewDbSchemaDialog.open(NewDbSchemaDialog.java:168)
	at com.magento.idea.magento2plugin.actions.generation.NewDbSchemaAction.actionPerformed(NewDbSchemaAction.java:48)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.lambda$performActionDumbAwareWith
```

**To Reproduce** (*)

**Expected behavior** (*)

**Screenshots**

**Please complete the following information:** (*)

 - OS: Windows 10 10.0
 - PhpStorm/Intellij version: 2021.2.2 Preview
 - Plugin Version: 4.0.0

**Additional context**

**Comment by bohdan-harniuk** [comment]
Hello, @Bashev !

Thank you for your report!
We'll try to deliver fix in the next bug-fix release!

Thank you!

**Comment by bohdan-harniuk** [comment]
Hello, @Bashev!

I've successfully reproduced this bug:

<img width="1009" alt="Screenshot 2021-11-30 at 21 05 45" src="https://user-images.githubusercontent.com/31848341/144120042-93a62e7b-4751-4d0e-8983-f082217892cd.png">
<img width="1170" alt="Screenshot 2021-11-30 at 21 06 31" src="https://user-images.githubusercontent.com/31848341/144120048-8168931d-337f-4abc-b1ab-f251051f761e.png">

The fix will be available in the next bug-fix release: 4.2.1.

Thanks, Bohdan
